backend/backend/core/chains.py

```diff
@@ -32,39 +31,43 @@ class AdjacentQueriesChain(object):
 
     class SearchQueryList(BaseModel):
         """List of search queries adjacent to a common query."""
+
         queries: List[str]
 
-    def __init__(self, model):
+    def __init__(self, models):
         prompt = ChatPromptTemplate.from_template(
             """
-            List {num_results} new adjacent search queries related to the query
-            '{query}', in the form of questions.
-
-            Your existing knowledge on the query is given below. The new search
-            queries must search for data not yet covered your existing
-            knowledge.
-
-            Output format: JSON list of strings according to schema:
-            {{
-                "queries": [
-                    "query1",
-                    "query2",
-                    ...
-                ]
-            }}
-
-            The existing knowledge is given below:
-            {knowledge}
-            """)
-        self.parser = PydanticOutputParser(
-            pydantic_object=AdjacentQueriesChain.SearchQueryList)
-        self.chain = prompt | model | self.parser
+List {num_results} new adjacent search queries related to the query
+'{query}', in the form of questions.
+
+Your existing knowledge on the query is given below. The new search
+queries must search for data not yet covered your existing
+knowledge.
+
+Output format: JSON list of strings according to schema:
+{{
+    "queries": [
+        "query1",
+        "query2",
+        ...
+    ]
+}}
+
+The existing knowledge is given below:
+{knowledge}
+            """
+        )
+        model = models.get_chat_model_json(
+            format=self.SearchQueryList.model_json_schema()
+        )
+        parser = PydanticOutputParser(pydantic_object=self.SearchQueryList)
+        self.chain = prompt | model | parser
 
     async def ainvoke(self, query, num_results, knowledge=None):
         logger.info(f"AdjacentQueriesChain: Getting queries for: {query}")
```

Comment:
The `ainvoke` method now takes a single argument `subject`, but it is unclear what this subject represents and how it relates to the rest of the code.